code/modules/vehicles/atv.dm

```diff
@@ -60,3 +62,41 @@
 				turret.pixel_x = 12
 				turret.pixel_y = 4
 				turret.layer = OBJ_LAYER
+
+/obj/vehicle/ridden/atv/attackby(obj/item/W as obj, mob/user as mob, params)
+	if(W.tool_behaviour == TOOL_WELDER && user.a_intent != INTENT_HARM)
+		if(obj_integrity < max_integrity)
+			if(W.use_tool(src, user, 0, volume=50, amount=1))
+				user.visible_message("<span class='notice'>[user] repairs some damage to [name].</span>", "<span class='notice'>You repair some damage to [src].</span>")
+				obj_integrity += min(10, max_integrity-obj_integrity)
+				if(obj_integrity == max_integrity)
+					to_chat(user, "<span class='notice'>It looks to be fully repaired now.</span>")
+		return TRUE
+	..()
+
+/obj/vehicle/ridden/atv/take_damage(damage_amount, damage_type = BRUTE, damage_flag = 0, sound_effect = 1, attack_dir, armour_penetration = 0)
+	. = ..()
+	if(obj_integrity <= 0)
+		explosion(src, -1, 0, 2, 4, flame_range = 3)
+	if(. && (obj_integrity > 0) && obj_integrity < (max_integrity * 0.5))
+		START_PROCESSING(SSobj, src)
```

Comment:
actually, apologies on this. move this to obj_break(), and set the integrity_failure var for the obj. (forgot we had that since it isn't used too often)